libraries/aws_vpc_subnet.rb

```diff
@@ -0,0 +1,105 @@
+# author: Matthew Dromazos
+
+require '_aws'
+
+class AwsVpcSubnet < Inspec.resource(1)
+  name 'aws_vpc_subnet'
+  desc 'This resource is used to test the attributes of a VPC subnet'
+  example "
+    describe aws_vpc_subnet(vpc_id: 'vpc-12345678', subnet_id: 'subnet-12345678') do
+      it { should exist }
+      its('cidr_block') { should eq '10.0.1.0/24' }
+    end
+  "
+
+  include AwsResourceMixin
+  attr_reader :vpc_id, :subnet_id, :cidr_block, :availability_zone, :available_ip_address_count,
+              :default_for_az, :map_public_ip_on_launch, :state, :ipv_6_cidr_block_association_set,
+              :assign_ipv_6_address_on_creation
+
+  def to_s
+    "VPC Subnet #{@subnet_id}"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:vpc_id, :subnet_id],
+      allowed_scalar_type: String,
+    )
+    # Make sure the params is not empty
+    if validated_params.empty?
+      raise ArgumentError, 'You must provide a vpc_id and subnet_id to aws_vpc_subnet.'
+    end
+
+    # Make sure the vpc_id parameter was specified and in the correct form.
+    if validated_params.key?(:vpc_id) && validated_params[:vpc_id] !~ /^vpc\-[0-9a-f]{8}/
+      raise ArgumentError, 'aws_vpc_subnet VPC ID must be in the format "vpc-" followed by 8 hexadecimal characters.'
+    end
+    # Make sure the subnet_id parameter was specified and in the correct form.
+    if validated_params.key?(:subnet_id) && validated_params[:subnet_id] !~ /^subnet\-[0-9a-f]{8}/
+      raise ArgumentError, 'aws_vpc_subnet Subnet ID must be in the format "subnet-" followed by 8 hexadecimal characters.'
+    end
+
+    validated_params
+  end
+
+  def fetch_from_aws
+    backend = AwsVpcSubnet::BackendFactory.create
+
+    # Transform into filter format expected by AWS
+    filters = []
```

Comment:
This is a really complicated way to make a two-element array.  I see this block copy-pasted a lot.  Can you think of ways to simplify it?